src/main/java/io/jenkins/plugins/opentelemetry/JenkinsOpenTelemetryPluginConfiguration.java

```diff
@@ -253,6 +258,34 @@ public StepPlugin findStepPluginOrDefault(@Nonnull String stepName, @Nullable St
         return data;
     }
 
+    /**
+     * @see io.opentelemetry.semconv.resource.attributes.ResourceAttributes#SERVICE_NAME
+     */
+    @CheckForNull
+    public String getServiceName() {
+        return (Strings.isNullOrEmpty(this.serviceName)) ? JenkinsOtelSemanticAttributes.JENKINS : this.serviceName;
+    }
+
+    @DataBoundSetter
+    public void setServiceName(String serviceName) {
```

Comment:
Is it `@Nullable` or `@NonNull`?

Reply:
`Nullable` thought the getter will return `jenkins` if so

Reply:
Do you mean to add the @Nullable annotation in the signature?

```suggestion
    public void setServiceName(@Nullable String serviceName) {
```